core/zip/inc/Compression.h

```diff
@@ -39,56 +39,71 @@ namespace ROOT {
+      kUseAnalysis = 404,
+      /// Use the recommended general-purpose setting; moderate read / write speed and compression ratio
+      kUseGeneralPurpose = 101,
+      /// Use the setting that results in the smallest files; very slow read and write
+      kUseSmallest = 207
+   };
+   enum class ELevel {
+      /// Some objects use this value to denote that the compression algorithm
+      /// should be inherited from the parent object
+      kInherit = -1,
+      // Compression level reserved for "uncompressed state"
+      kUncompressed = 0,
+      // Compression level reserved when we are not sure what to use (1 is for the fastest compression)
+      kUseMin = 1,
+      kDefaultZLIB = 1,
+      kDefaultLZ4 = 4,
+      kDefaultOld = 6,
+      kDefaultLZMA = 7
+   };
+   enum class EAlgorithm {
+      /// Some objects use this value to denote that the compression algorithm
+      /// should be inherited from the parent object (e.g., TBranch should get the algorithm from the TTree)
+      kInherit = -1,
+      /// Use the global compression algorithm
+      kUseGlobal = 0,
+      /// Use ZLIB compression
+      kZLIB,
+      /// Use LZMA compression
+      kLZMA,
+      /// Use the old compression algorithm
+      kOldCompressionAlgo,
+      /// Use LZ4 compression
+      kLZ4,
+      /// Undefined compression algorithm (must be kept the last of the list in case a new algorithm is added).
+      kUndefined
+   };
 };
 
-enum ECompressionSetting {
-   /// Use the global compression setting for this process; may be affected by rootrc.
-   kUseGlobalCompressionSetting = 0,
-   /// Use the compile-time default setting
-   kUseCompiledDefaultCompressionSetting = 404,
-   /// Use the default analysis setting; fast reading but poor compression ratio
-   kUseAnalysisCompressionSetting = 404,
-   /// Use the recommended general-purpose setting; moderate read / write speed and compression ratio
-   kUseGeneralPurposeCompressionSetting = 101,
-   /// Use the setting that results in the smallest files; very slow read and write
-   kUseSmallestCompressionSetting = 207
+enum ECompressionAlgorithm {
+   // Deprecated name, do *not* use:
+   kInherit = static_cast<int>(RCompressionSetting::EAlgorithm::kInherit),
+   /// Deprecated name, do *not* use:
+   kUseGlobalSetting = static_cast<int>(RCompressionSetting::EAlgorithm::kUseGlobal),
+   /// Deprecated name, do *not* use:
+   kZLIB = static_cast<int>(RCompressionSetting::EAlgorithm::kZLIB),
+   /// Deprecated name, do *not* use:
+   kLZMA = static_cast<int>(RCompressionSetting::EAlgorithm::kLZMA),
+   /// Deprecated name, do *not* use:
+   kOldCompressionAlgo = static_cast<int>(RCompressionSetting::EAlgorithm::kOldCompressionAlgo),
+   /// Deprecated name, do *not* use:
+   kLZ4 = static_cast<int>(RCompressionSetting::EAlgorithm::kLZ4),
+   /// Deprecated name, do *not* use:
+   kUndefinedCompressionAlgorithm = static_cast<int>(RCompressionSetting::EAlgorithm::kUndefined)
```

Comment:
if I understand correctly, enum values in `RCompressionSetting::EAlgorithm` _must_ be listed in the same exact order as they were in `ECompressionAlgorithm` in order to avoid potentially changing the behavior of code that was relying on the `int` values of `ECompressionAlgorithm`. This is currently the case, but if I'm right then it would be good to have a comment in `RCompressionSetting::EAlgorithm` that shouts at people not to change the order of the values.

Reply:
int numbers are exactly the same (https://github.com/root-project/root/blob/4aa3e31999ff3569ce0a3e88b887102e590961bc/core/zip/inc/Compression.h#L34).
only difference that by default enumerator starts with 0 (like it was in old case), now we add kInherit = -1, and use exactly the same order as it was before. RCompressionSetting::EAlgorithm will be deprecated but if you check it has exactly the same values as in the old case.